unit_tests/src/test_resource_administration.cpp

```diff
@@ -16,199 +16,191 @@ TEST_CASE("resource administration")
 
     adm::resource_registration_info repl_info;
     repl_info.resource_name = "unit_test_repl";
-    repl_info.resource_type = adm::resource_type::replication.data();
+    repl_info.resource_type = adm::resource_type::replication;
 
     char host_name[64] {};
     REQUIRE(gethostname(host_name, sizeof(host_name)) == 0);
 
     adm::resource_registration_info ufs_info;
     ufs_info.resource_name = "unit_test_ufs0";
-    ufs_info.resource_type = adm::resource_type::unixfilesystem.data();
+    ufs_info.resource_type = adm::resource_type::unixfilesystem;
     ufs_info.host_name = host_name;
     ufs_info.vault_path = "/tmp";
 
     load_client_api_plugins();
 
-    auto conn_pool = irods::make_connection_pool();
-    auto conn = conn_pool->get_connection();
+    irods::experimental::client_connection conn;
 
     irods::at_scope_exit remove_resources{[&conn, &repl_info, &ufs_info] {
-        adm::client::remove_resource(conn, repl_info.resource_name);
-        adm::client::remove_resource(conn, ufs_info.resource_name);
+        try {
+            CHECK_NOTHROW(adm::client::remove_resource(conn, repl_info.resource_name));
```

Comment:
I was under the impression that CHECK_NOTHROW and friends handled the exceptions or lack thereof which would render the try-catch unnecessary. Is that not the case?

I guess it doesn't matter since this is cleanup code, so it doesn't hurt to be safe.

Same thing for 41

Reply:
I wasn't sure and I didn't want to run the risk of the other call not executing.

I'll verify and tweak.

Reply:
This comment from the Catch2 docs indicates that the `CHECK_NOTHROW` macro works as we suspect it.
```
The CHECK family are equivalent but execution continues in the same test case even if the assertion fails. This is useful if you have a series of essentially orthogonal assertions and it is useful to see all the results rather than stopping at the first failure.
```
I think we're covered.

Reply:
I've verified through manual testing that the try-catch blocks around the `CHECK_NOTHROW` macros aren't necessary.

The code has been updated.